Codeblock breaks code generation

```java
@Mcp.Tool("""
        Returns example of Helidon CLI command to create Helidon quickstart example with provided projectName
        """)
List<McpToolContent> getInitHelidonSeProjectWithCliCmd(String projectName) {
    return List.of(McpToolContents.textContent("test"));
}
```


```java
[ERROR] COMPILATION ERROR : 
[INFO] -------------------------------------------------------------
[ERROR] /home/daniel/idp/ora/helidon/helidon-mcp/examples/calendar-application/calendar-declarative/target/generated-sources/annotations/io/helidon/extensions/mcp/examples/calendar/declarative/McpCliServer__McpServer.java:[94,20] unclosed string literal
[ERROR] /home/daniel/idp/ora/helidon/helidon-mcp/examples/calendar-application/calendar-declarative/target/generated-sources/annotations/io/helidon/extensions/mcp/examples/calendar/declarative/McpCliServer__McpServer.java:[95,13] unclosed string literal
```

<img width="1201" height="219" alt="Image" src="https://github.com/user-attachments/assets/9d1e0de5-8f95-4d97-96e0-a698c4aea59e" />

--- a/codegen/src/main/java/io/helidon/extensions/mcp/codegen/McpCompletionCodegen.java
+++ b/codegen/src/main/java/io/helidon/extensions/mcp/codegen/McpCompletionCodegen.java
@@ -70,9 +70,9 @@ class McpCompletionCodegen {
         builder.name("reference")
                 .addAnnotation(Annotations.OVERRIDE)
                 .returnType(TypeNames.STRING)
-                .addContent("return \"")
-                .addContent(reference)
-                .addContentLine("\";");
+                .addContent("return ")
+                .addContentLiteral(reference)
+                .addContentLine(";");
     }
 
     private void addCompletionReferenceTypeMethod(Method.Builder builder, EnumValue referenceType) {

--- a/codegen/src/main/java/io/helidon/extensions/mcp/codegen/McpPromptCodegen.java
+++ b/codegen/src/main/java/io/helidon/extensions/mcp/codegen/McpPromptCodegen.java
@@ -76,25 +76,25 @@ class McpPromptCodegen {
         builder.name("name")
                 .addAnnotation(Annotations.OVERRIDE)
                 .returnType(TypeNames.STRING)
-                .addContent("return \"")
-                .addContent(name)
-                .addContentLine("\";");
+                .addContent("return ")
+                .addContentLiteral(name)
+                .addContentLine(";");
     }
 
     private void addPromptDescriptionMethod(Method.Builder builder, String description) {
         builder.name("description")
                 .addAnnotation(Annotations.OVERRIDE)
                 .returnType(TypeNames.STRING)
-                .addContent("return \"")
-                .addContent(description)
-                .addContentLine("\";");
+                .addContent("return ")
+                .addContentLiteral(description)
+                .addContentLine(";");
     }
 
     private void addPromptMethod(Method.Builder builder, ClassModel.Builder classModel, TypedElementInfo element) {
         List<String> parameters = new ArrayList<>();
         TypeName returnType = element.signature().type();
         String role = element.findAnnotation(MCP_ROLE)
-                .flatMap(annotation -> annotation.value())
+                .flatMap(Annotation::value)
                 .orElse("ASSISTANT");
 
         builder.name("prompt")
@@ -115,9 +115,9 @@ class McpPromptCodegen {
                 builder.addContent(param.typeName().classNameWithEnclosingNames())
                         .addContent(" ")
                         .addContent(param.elementName())
-                        .addContent(" = request.arguments().get(\"")
-                        .addContent(param.elementName())
-                        .addContentLine("\").asString().orElse(\"\");");
+                        .addContent(" = request.arguments().get(")
+                        .addContentLiteral(param.elementName())
+                        .addContentLine(").asString().orElse(\"\");");
                 continue;
             }
             throw new CodegenException(String.format("Wrong parameter type for method: %s. Supported types are: %s, or String.",
@@ -181,9 +181,9 @@ class McpPromptCodegen {
                     .addContent(MCP_PROMPT_ARGUMENT)
                     .addContentLine(".builder();")
                     .addContent(builderName)
-                    .addContent(".name(\"")
-                    .addContent(param.elementName())
-                    .addContentLine("\");")
+                    .addContent(".name(")
+                    .addContentLiteral(param.elementName())
+                    .addContentLine(");")
                     .addContent(builderName)
                     .addContentLine(".required(true);");
 
@@ -192,15 +192,15 @@ class McpPromptCodegen {
             if (param.hasAnnotation(MCP_DESCRIPTION)) {
                 String description = param.annotation(MCP_DESCRIPTION).value().orElse("");
                 builder.addContent(builderName)
-                        .addContent(".description(\"")
-                        .addContent(description)
-                        .addContentLine("\");");
+                        .addContent(".description(")
+                        .addContentLiteral(description)
+                        .addContentLine(");");
                 continue;
             }
             builder.addContent(builderName)
-                    .addContent(".description(\"")
-                    .addContent(param.elementName())
-                    .addContentLine("\");");
+                    .addContent(".description(")
+                    .addContentLiteral(param.elementName())
+                    .addContentLine(");");
         }
         builder.addContent("return ")
                 .addContent(List.class)

--- a/codegen/src/main/java/io/helidon/extensions/mcp/codegen/McpResourceCodegen.java
+++ b/codegen/src/main/java/io/helidon/extensions/mcp/codegen/McpResourceCodegen.java
@@ -130,27 +130,27 @@ class McpResourceCodegen {
         builder.name("name")
                 .addAnnotation(Annotations.OVERRIDE)
                 .returnType(TypeNames.STRING)
-                .addContent("return \"")
-                .addContent(name)
-                .addContentLine("\";");
+                .addContent("return ")
+                .addContentLiteral(name)
+                .addContentLine(";");
     }
 
     private void addResourceDescriptionMethod(Method.Builder builder, String description) {
         builder.name("description")
                 .addAnnotation(Annotations.OVERRIDE)
                 .returnType(TypeNames.STRING)
-                .addContent("return \"")
-                .addContent(description)
-                .addContentLine("\";");
+                .addContent("return ")
+                .addContentLiteral(description)
+                .addContentLine(";");
     }
 
     private void addResourceUriMethod(Method.Builder builder, String uri) {
         builder.name("uri")
                 .addAnnotation(Annotations.OVERRIDE)
                 .returnType(TypeNames.STRING)
-                .addContent("return \"")
-                .addContent(uri)
-                .addContentLine("\";");
+                .addContent("return ")
+                .addContentLiteral(uri)
+                .addContentLine(";");
     }
 
     private void addResourceMediaTypeMethod(Method.Builder builder, String mediaTypeContent) {
@@ -159,9 +159,9 @@ class McpResourceCodegen {
                 .returnType(HELIDON_MEDIA_TYPE)
                 .addContent("return ")
                 .addContent(HELIDON_MEDIA_TYPES)
-                .addContent(".create(\"")
-                .addContent(mediaTypeContent)
-                .addContentLine("\");");
+                .addContent(".create(")
+                .addContentLiteral(mediaTypeContent)
+                .addContentLine(");");
     }
 
     private void addResourceMethod(Method.Builder builder, String uri, ClassModel.Builder classModel, TypedElementInfo element) {
@@ -186,9 +186,9 @@ class McpResourceCodegen {
                     parameters.add(parameter.elementName());
                     builder.addContent("String encoded_")
                             .addContent(parameter.elementName())
-                            .addContent(" = request.parameters().get(\"")
-                            .addContent(parameter.elementName())
-                            .addContentLine("\").asString().orElse(\"\");");
+                            .addContent(" = request.parameters().get(")
+                            .addContentLiteral(parameter.elementName())
+                            .addContentLine(").asString().orElse(\"\");");
                     builder.addContent("String ")
                             .addContent(parameter.elementName())
                             .addContent(" = ")
@@ -234,9 +234,9 @@ class McpResourceCodegen {
         builder.name("uri")
                 .addAnnotation(Annotations.OVERRIDE)
                 .returnType(TypeNames.STRING)
-                .addContent("return \"")
-                .addContent(uri)
-                .addContentLine("\";");
+                .addContent("return ")
+                .addContentLiteral(uri)
+                .addContentLine(";");
     }
 
     private void addSubscriberMethod(Method.Builder builder, TypedElementInfo element) {

--- a/codegen/src/main/java/io/helidon/extensions/mcp/codegen/McpToolCodegen.java
+++ b/codegen/src/main/java/io/helidon/extensions/mcp/codegen/McpToolCodegen.java
@@ -167,27 +167,27 @@ class McpToolCodegen {
                 parameters.add(param.elementName());
                 builder.addContent("var ")
                         .addContent(param.elementName())
-                        .addContent(" = request.arguments().get(\"")
-                        .addContent(param.elementName())
-                        .addContentLine("\").asString().orElse(\"\");");
+                        .addContent(" = request.arguments().get(")
+                        .addContentLiteral(param.elementName())
+                        .addContentLine(").asString().orElse(\"\");");
                 continue;
             }
             if (isBoolean(param.typeName())) {
                 parameters.add(param.elementName());
                 builder.addContent("boolean ")
                         .addContent(param.elementName())
-                        .addContent(" = request.arguments().get(\"")
-                        .addContent(param.elementName())
-                        .addContentLine("\").asBoolean().orElse(false);");
+                        .addContent(" = request.arguments().get(")
+                        .addContentLiteral(param.elementName())
+                        .addContentLine(").asBoolean().orElse(false);");
                 continue;
             }
             if (isNumber(param.typeName())) {
                 parameters.add(param.elementName());
                 builder.addContent("var ")
                         .addContent(param.elementName())
-                        .addContent(" = request.arguments().get(\"")
-                        .addContent(param.elementName())
-                        .addContent("\").as")
+                        .addContent(" = request.arguments().get(")
+                        .addContentLiteral(param.elementName())
+                        .addContent(").as")
                         .addContent(param.typeName().className())
                         .addContentLine("().orElse(null);");
                 continue;
@@ -198,18 +198,18 @@ class McpToolCodegen {
                 parameters.add(param.elementName());
                 builder.addContent("var ")
                         .addContent(param.elementName())
-                        .addContent(" = toList(request.arguments().get(\"")
-                        .addContent(param.elementName())
-                        .addContentLine("\").asList().orElse(null));");
+                        .addContent(" = toList(request.arguments().get(")
+                        .addContentLiteral(param.elementName())
+                        .addContentLine(").asList().orElse(null));");
                 continue;
             }
             parameters.add(param.elementName());
             builder.addContent(param.typeName().classNameWithEnclosingNames())
                     .addContent(" ")
                     .addContent(param.elementName())
-                    .addContent(" = request.arguments().get(\"")
-                    .addContent(param.elementName())
-                    .addContent("\").as(")
+                    .addContent(" = request.arguments().get(")
+                    .addContentLiteral(param.elementName())
+                    .addContent(").as(")
                     .addContent(param.typeName())
                     .addContentLine(".class).orElse(null);");
         }
@@ -247,18 +247,18 @@ class McpToolCodegen {
         builder.name("name")
                 .addAnnotation(Annotations.OVERRIDE)
                 .returnType(TypeNames.STRING)
-                .addContent("return \"")
-                .addContent(name)
-                .addContentLine("\";");
+                .addContent("return ")
+                .addContentLiteral(name)
+                .addContentLine(";");
     }
 
     private void addToolDescriptionMethod(Method.Builder builder, String description) {
         builder.name("description")
                 .addAnnotation(Annotations.OVERRIDE)
                 .returnType(TypeNames.STRING)
-                .addContent("return \"")
-                .addContent(description)
-                .addContentLine("\";");
+                .addContent("return ")
+                .addContentLiteral(description)
+                .addContentLine(";");
     }
 
     private void addToolAnnotationsMethod(Method.Builder builder, Annotation toolAnnotation) {
@@ -266,9 +266,9 @@ class McpToolCodegen {
                 .addAnnotation(Annotations.OVERRIDE)
                 .returnType(OPTIONAL_TOOL_ANNOTATIONS)
                 .addContentLine("var builder = McpToolAnnotations.builder();")
-                .addContent("builder.title(\"")
-                .addContent(toolAnnotation.stringValue("title").orElse(""))
-                .addContentLine("\")")
+                .addContent("builder.title(")
+                .addContentLiteral(toolAnnotation.stringValue("title").orElse(""))
+                .addContentLine(")")
                 .increaseContentPadding()
                 .addContent(".readOnlyHint(")
                 .addContent(toolAnnotation.booleanValue("readOnlyHint").orElse(false).toString())

--- a/tests/declarative/src/main/java/io/helidon/extensions/mcp/tests/declarative/McpPromptsServer.java
+++ b/tests/declarative/src/main/java/io/helidon/extensions/mcp/tests/declarative/McpPromptsServer.java
@@ -96,4 +96,16 @@ class McpPromptsServer {
     McpPromptResult prompt9(McpPromptRequest request) {
         return McpPromptResult.builder().addTextContent(PROMPT_CONTENT).build();
     }
+
+    @Mcp.Prompt("""
+            Description code block
+            """)
+    String prompt10(McpRequest request) {
+        return PROMPT_CONTENT;
+    }
+
+    @Mcp.Prompt("first line\n second line\n")
+    String prompt11(McpRequest request) {
+        return PROMPT_CONTENT;
+    }
 }

--- a/tests/declarative/src/main/java/io/helidon/extensions/mcp/tests/declarative/McpResourcesServer.java
+++ b/tests/declarative/src/main/java/io/helidon/extensions/mcp/tests/declarative/McpResourcesServer.java
@@ -85,4 +85,23 @@ class McpResourcesServer {
     McpResourceResult resource6(McpResourceRequest request) {
         return McpResourceResult.builder().addTextContent(RESOURCE_CONTENT).build();
     }
+
+    @Mcp.Resource(
+            uri = "resource5",
+            mediaType = RESOURCE_MEDIA_TYPE,
+            description = """
+                    Description code block
+                    """)
+    McpResourceResult resource7(McpResourceRequest request) {
+        return McpResourceResult.create(RESOURCE_CONTENT);
+    }
+
+    @Mcp.Resource(
+            uri = "resource5",
+            mediaType = RESOURCE_MEDIA_TYPE,
+            description = "first line\n"
+                    + "second line")
+    McpResourceResult resource8(McpResourceRequest request) {
+        return McpResourceResult.create(RESOURCE_CONTENT);
+    }
 }

--- a/tests/declarative/src/main/java/io/helidon/extensions/mcp/tests/declarative/McpToolsServer.java
+++ b/tests/declarative/src/main/java/io/helidon/extensions/mcp/tests/declarative/McpToolsServer.java
@@ -161,6 +161,18 @@ class McpToolsServer {
                 .build();
     }
 
+    @Mcp.Tool("""
+            Tool description block
+            """)
+    McpToolResult tool19(McpRequest request) {
+        return McpToolResult.create(TOOL_CONTENT);
+    }
+
+    @Mcp.Tool("first line\n" + "second line\n")
+    McpToolResult tool20(McpRequest request) {
+        return McpToolResult.create(TOOL_CONTENT);
+    }
+
     @JsonSchema.Schema
     public static class Foo {
         public String foo;

--- a/tests/declarative/src/test/java/io/helidon/extensions/mcp/tests/declarative/AbstractLangchain4jPromptsServerTest.java
+++ b/tests/declarative/src/test/java/io/helidon/extensions/mcp/tests/declarative/AbstractLangchain4jPromptsServerTest.java
@@ -46,9 +46,12 @@ abstract class AbstractLangchain4jPromptsServerTest {
     }
 
     @Test
-    void listPrompts() {
+    void testListPrompt() {
         List<McpPrompt> prompts = client.listPrompts();
-        assertThat(prompts.size(), is(13));
+        assertThat(prompts.size(), is(15));
+
+        assertThat(prompts.get(13).description(), is("Description code block\n"));
+        assertThat(prompts.get(14).description(), is("first line\n second line\n"));
     }
 
     @ParameterizedTest

--- a/tests/declarative/src/test/java/io/helidon/extensions/mcp/tests/declarative/AbstractLangchain4jResourcesServerTest.java
+++ b/tests/declarative/src/test/java/io/helidon/extensions/mcp/tests/declarative/AbstractLangchain4jResourcesServerTest.java
@@ -44,8 +44,14 @@ abstract class AbstractLangchain4jResourcesServerTest {
 
     @Test
     void listResources() {
-        List<McpResource> resources = client.listResources();
-        assertThat(resources.size(), is(7));
+        List<McpResource> list = client.listResources();
+        assertThat(list.size(), is(9));
+
+        McpResource resource = list.get(7);
+        assertThat(resource.description(), is("Description code block\n"));
+
+        McpResource resource2 = list.get(8);
+        assertThat(resource2.description(), is("first line\nsecond line"));
     }
 
     @ParameterizedTest

--- a/tests/declarative/src/test/java/io/helidon/extensions/mcp/tests/declarative/AbstractLangchain4jToolsServerTest.java
+++ b/tests/declarative/src/test/java/io/helidon/extensions/mcp/tests/declarative/AbstractLangchain4jToolsServerTest.java
@@ -17,6 +17,7 @@
 package io.helidon.extensions.mcp.tests.declarative;
 
 import dev.langchain4j.agent.tool.ToolExecutionRequest;
+import dev.langchain4j.agent.tool.ToolSpecification;
 import dev.langchain4j.mcp.client.McpClient;
 import org.junit.jupiter.api.AfterAll;
 import org.junit.jupiter.api.Test;
@@ -41,7 +42,13 @@ abstract class AbstractLangchain4jToolsServerTest {
     @Test
     void testListTools() {
         var list = client.listTools();
-        assertThat(list.size(), is(19));
+        assertThat(list.size(), is(21));
+
+        ToolSpecification tool1 = list.get(19);
+        assertThat(tool1.description(), is("Tool description block\n"));
+
+        ToolSpecification tool2 = list.get(20);
+        assertThat(tool2.description(), is("first line\nsecond line\n"));
     }
 
     @ParameterizedTest

